shell/packages/sandstorm-ui-powerbox/powerbox-client.js

```diff
@@ -193,133 +133,103 @@ SandstormPowerboxRequest = class SandstormPowerboxRequest {
     });
   }
 
-  selectApiToken(apiTokenCard) {
-    Meteor.call("getViewInfoForApiToken", apiTokenCard._id, (err, result) => {
-      if (err) {
-        console.log(err);
-        this._error.set(err.toString());
-      } else {
-        const viewInfo = result;
-        this.annotateViewInfo(viewInfo);
-        this._selectedProvider.set({
-          type: "frontendref-uiview",
-          grainId: apiTokenCard.grainId,
-          apiToken: apiTokenCard._id,
-          title: apiTokenCard.title,
-          templateName: "powerboxProviderUiView",
-          templateData: () => {
-            return {
-              _id: apiTokenCard._id,
-              title: apiTokenCard.title,
-              appTitle: apiTokenCard.appTitle,
-              iconSrc: apiTokenCard.iconSrc,
-              lastUsed: apiTokenCard.lastUsed,
-              viewInfo: viewInfo,
-              onComplete: (roleAssignment) => {
-                this.completeUiView(roleAssignment);
-              },
-            };
-          },
-        });
+  filteredCardData() {
+    const cards = PowerboxOptions.find({ requestId: this._requestId }).map(cardData => {
+      // Use ID as title if we don't find anything better.
+      cardData.title = cardData._id;
+
+      if (cardData.grainId) {
+        this.extendWithGrainInfo(cardData);
```

Comment:
I find this logic a little hard to follow, in that it doesn't tell me anything about the universe of fields I can expect to find in the eventual return value.  I can tell from the comments in the serverside publish for `PowerboxOptions` what values come from there, but it's not clear at a glance what `extendWithGrainInfo` will add, remove, or change, or under what conditions I can expect those values to be present or absent.

The previous code was a bit more explicit here; the final object structure was all in one place.  Documenting which fields are guaranteed to be present in the value returned by `filteredCardData()` would be a welcome addition.

Maybe these cardData instances are large enough that they deserve a class extraction, which would give some clearer guarantees about what data this type contains?